src/Compilers/Server/VBCSCompilerTests/ClientConnectionHandlerTests.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO.Pipes;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CommandLine;
+using Moq;
+using Roslyn.Test.Utilities;
+using Xunit;
+using System.Runtime.InteropServices;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using static Microsoft.CodeAnalysis.CommandLine.BuildResponse;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CompilerServer.UnitTests
+{
+    public class ClientConnectionHandlerTests
+    {
+        [Fact]
+        public async Task ThrowDuringBuild()
+        {
+            var compilerServerHost = new TestableCompilerServerHost(delegate { throw new Exception(); });
+            var clientConnectionHandler = new ClientConnectionHandler(compilerServerHost);
+            var clientConnection = new TestableClientConnection()
+            {
+                ReadBuildRequestFunc = _ => Task.FromResult(ProtocolUtil.EmptyCSharpBuildRequest),
+            };
+            var completionData = await clientConnectionHandler.ProcessAsync(Task.FromResult<IClientConnection>(clientConnection)).ConfigureAwait(false);
+            Assert.Equal(CompletionData.RequestError, completionData);
+        }
+
+        [Fact]
+        public async Task ThrowReadingRequest()
+        {
+            var compilerServerHost = new TestableCompilerServerHost(delegate { throw new Exception(); });
```

Comment:
>(delegate { throw new Exception(); } [](start = 67, length = 36)

Perhaps make this actually do something, so the test will actually demonstrate that it's the below exception that failed, rather than possibly this one?